client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx

```diff
@@ -88,14 +91,36 @@ export const CodeMonitorForm: React.FunctionComponent<CodeMonitorFormProps> = ({
                     mergeMap(() =>
                         onSubmit(currentCodeMonitorState).pipe(
                             startWith(LOADING),
-                            catchError(error => [asError(error)])
+                            catchError(error => [asError(error)]),
+                            tap(successOrError => {
+                                if (!isErrorLike(successOrError) && successOrError !== LOADING) {
+                                    history.push('/code-monitoring')
+                                }
+                            })
                         )
                     )
                 ),
-            [onSubmit, currentCodeMonitorState]
+            [onSubmit, currentCodeMonitorState, history]
         )
     )
 
+    const initialCodeMonitor = useMemo(() => codeMonitor, [codeMonitor])
+
+    // Determine whether the form has changed. If there was no intial state (i.e. we're creating a monitor), always return
+    // true.
+    const hasChangedFields = useMemo(
+        () => (codeMonitor ? !isEqual(initialCodeMonitor, currentCodeMonitorState) : true),
+        [initialCodeMonitor, codeMonitor, currentCodeMonitorState]
+    )
+
+    const onCancel = useCallback(() => {
+        if (hasChangedFields) {
+            if (confirm('Leave page? All unsaved changes will be lost.')) {
```

Comment:
```suggestion
            if (window.confirm('Leave page? All unsaved changes will be lost.')) {
```
Can we more explicitly use `window.confirm`?